### PR TITLE
unix: `uv_pipe_init` should exports error

### PR DESCRIPTION
**Pull request by yorkie**
Related lines:
1. https://github.com/joyent/libuv/blob/7e68e6d895a9007736e53b592192fd9620a34ac6/src/unix/stream.c#L77
2. https://github.com/joyent/libuv/blob/master/src/win/pipe.c#L92

### CODE REVIEW

File: src/unix/pipe.c [hunk 1]
```diff
@@ -36,7 +36,7 @@ int uv_pipe_init(uv_loop_t* loop, uv_pipe_t* handle, int ipc) {
   handle->connect_req = NULL;
   handle->pipe_fname = NULL;
   handle->ipc = ipc;
-  return 0;
+  return errno;
```
Comment (indutny) [inline src/unix/pipe.c:39]: Hm... why?

**Comment by indutny** [comment]
Sorry, it doesn't look like a right thing

File: src/unix/pipe.c (hunk 1 above)
Comment (yorkie) [inline src/unix/pipe.c:39]: Did you check this branch: https://github.com/joyent/libuv/blob/7e68e6d895a9007736e53b592192fd9620a34ac6/src/unix/stream.c#L77?

Comment (indutny) [inline src/unix/pipe.c:39]: And? :) It is only for `emfile_fd` trick, if it fails - everything is still fine. See https://github.com/joyent/libuv/blob/7e68e6d895a9007736e53b592192fd9620a34ac6/src/unix/tcp.c#L31 for reference

Comment (yorkie) [inline src/unix/pipe.c:39]: when `uv__stream_init` fails, the `errno` would be assigned with the specified non-zero value.
Actually, check this https://github.com/joyent/node/blob/master/src/pipe_wrap.cc#L134 in node, we should export an error for this.

**Comment by indutny** [comment]
Sorry, you're doing wrong conclusions from the comments you see.

**Comment by yorkie** [comment]
I see, it's my fault, sorry!

**Pull request closed without merging**